src/observer/sitemap.py

```diff
@@ -0,0 +1,53 @@
+from io import StringIO
+from observer import models, utils
+from django.http import HttpResponse  # , StreamingHttpResponse
+
+xml_doc_header = """<?xml version="1.0" encoding="UTF-8"?>
+<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd">\n"""
+
+xml_doc_footer = """</urlset>"""
+
+def url_elem(loc_str, mod_str):
+    return "  <url>\n    <loc>" + loc_str + "</loc>\n    <lastmod>" + mod_str + "</lastmod>\n  </url>\n"
+
+def render(data_pair_list):
+    "returns a generator function that first yields the header, then each body item, then the footer"
+    yield xml_doc_header
+    for loc_str, mod_str in data_pair_list:
+        yield url_elem(loc_str, mod_str)
+    yield xml_doc_footer
+
+def coerce(item):
+    if isinstance(item, tuple):
+        return item # assume data is ready to go as-is
+    if isinstance(item, dict):
+        return (item['url'], utils.ymdhms(item['last-modified']))
+    if isinstance(item, models.Article):
+        return (models.content_url(item), utils.ymdhms(item.datetime_version_published))
+    if isinstance(item, models.PressPackage):
+        return (models.content_url(item), utils.ymdhms(item.updated or item.published))
+    if isinstance(item, models.Content):
+        return (models.content_url(item), utils.ymdhms(item.datetime_updated or item.datetime_published))
+    raise ValueError("item %r cannot be coerced to a pair of (url, last-modified) values" % (item,))
+
+def _format_report(report, context):
+    "generates a `sitemap.xml` from the given `report` and `context` data, returning an HttpResponse."
```

Comment:
generates a `sitemap.xml` document from the given `report` and its `context` data, returning a generator